tests/lib/rules/operator-assignment.js

```diff
@@ -398,6 +398,20 @@ ruleTester.run("operator-assignment", rule, {
         output: "foo= foo+(+bar===baz)", // tokens cannot be adjacent, but the right side will be parenthesised
         options: ["never"],
         errors: UNEXPECTED_OPERATOR_ASSIGNMENT
-    }]
+    },
+
+    // Optional chaining
+    {
+        code: "(obj?.a).b = (obj?.a).b + y",
+        output: null,
+        errors: EXPECTED_OPERATOR_ASSIGNMENT
+    },
+    {
+        code: "obj.a = obj?.a + b",
```

Comment:
Should `no-self-assign` also report similar cases?

It uses `astUtils.isSameReference` so it will report `a.b.c = (a?.b).c;`

But it checks first if they're both `MemberExpression`, so it will not report `a.b = a?.b;`

Reply:
Thank you! I updated `no-self-assign` rule to catch similar cases.